### ISSUE TITLE
Nullpointer exception for some BigQuery results

### ISSUE THREAD

**Issue by johanneslumpe** [metabase/metabase#3788]
First of all - a big thank you for metabase. It's an awesome project. I've tried to use the BigQuery `GENERATE_DATE_ARRAY` function to generate a result set of dates, which lead to a Nullpointer exception error in metabase. 

The metabase version we are using is:
```
v0.21.0-snapshot
Built on 2016-11-14
Branch: master
Hash: 36e3d8a
```
It has patched support for a `--nolegacy` flag, to allow using standard SQL in BigQuery. Running the following example query - taken directly from the BigQuery docs - works fine in BigQuery, but returns said exception in metabase:

`SELECT GENERATE_DATE_ARRAY('2016-10-05', '2016-10-08') AS example;`

I don't fully understand why the result would break metabase, because it simply returns multiple rows with a single column. Just to make sure that it wasn't BigQuery automatically unnesting the results, I also tried to manually force separate rows using `UNNEST`, but that yielded the same error.

Any thoughts/help on this are/is greatly appreciated.

**Comment by thekantian** [comment]
Using the latest docker container for 0.20.3 for Metabase.

I think I've got a related issue - when syncing a new table I've added to an existing BigQuery database, (the table has "DATE" values in 2 columns) I'm getting an error in the metabase logs like:

```
:cause Value does not match schema: {:fields #{{:base-type (not ("Valid type" nil))} {:base-type (not ("Valid type" nil))}}}
```

The table does not get added to the queryable tables list in admin, and queries using the table fail with the NullPointerException in the UI. In the BigQuery UI the query will work fine. It seems to throw the above error in the when it encounters a DATE field in bigquery. So I think it's probably related to the above.

Is there any way to work around this that you guys can think of? Thanks so much!

**Comment by camsaul** [comment]
@johanneslumpe @thekantian I think this is an issue with missing mappings for BigQuery `DATE` and `DATETIME` types to internal Metabase types.

@gabrielrcouto opened PR #3692 to address this so hopefully a fix for this should be available in our upcoming release.

**Comment by johanneslumpe** [comment]
@camsaul awesome to know! Thanks! Can't wait ;)

**Comment by thekantian** [comment]
Awesome, thanks so much guys happy to help test if we have a docker image with the newer release…

**Comment by felipegasparini** [comment]
Guys, just one thing:

`SELECT GENERATE_DATE_ARRAY('2016-10-05', '2016-10-08') AS example;`

the function GENERATE_DATE_ARRAY actually returns an ARRAY of DATEs , i.e. it doesn't return a result set with N rows, but returns a single row with N values in the column example.

You can check this behavior with the query:

`SELECT GENERATE_DATE_ARRAY('2016-10-05', '2016-10-08') AS example, 1 as row
UNION ALL
SELECT NULL AS example, 2 as row;
`

I don't know if Metabase already supports the BQ ARRAY Type (https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#array-type) but I think it is also missing! =)

**Comment by johanneslumpe** [comment]
@felipegasparini as a workaround you can `UNNEST` that array :)

**Comment by camsaul** [comment]
@felipegasparini Metabase doesn't currently have any sort of special support like automatic unnesting for arrays , but it's something we're looking at adding in the future.

Related: #2974 (support MongoDB arrays)

**Comment by camsaul** [comment]
Fixed by #3933
